### PR TITLE
Fix syncing

### PR DESCRIPTION
**Pull request by bedeoverend**
Fixes up syncing, but removes a lot of stuff. Rather than sending messages back and forth to try keep the editor and viewer in sync, it now just uses Polymer's two-way binding. Means a lot less work for us and allows us to remove most of the logic.

Also removes the `add` / `remove` / `swap` functions on the view collection. @seaneking what do you think of this? 

There's no point to them unless we want to expose an interface for elements consumers to use, which I'm happy to do, but just wanted to trim it back to barest minimum to start with. I think when we were initially setting it up I was thinking the editor would have to be a separate thing living outside of the viewer and sending commands back to it, but while we're using Polymer, may as well make use of it's bindings.

### CODE REVIEW

File: src/simpla-collection.html [hunk 1]
```diff
@@ -45,7 +45,7 @@
          */
         items: {
           type: Array,
-          value: []
+          value: () => []
```
Comment (bedeoverend) [inline src/simpla-collection.html:48]: @seaneking just FYI need to watch out for this

File: src/simpla-collection-editor.html [hunk 2]
```diff
@@ -331,42 +336,15 @@
...
-        let itemData = Polymer.dom(e).rootTarget
-              .closest(`[${ITEM_DATA_ATTR}]`)
-              .getAttribute(ITEM_DATA_ATTR),
-            item = JSON.parse(itemData);
-
-        this.fire('collection-item-edit', {
-          action: 'remove',
-          args: [ item.id ]
-        });
+        this.splice('items', e.model.index, 1);
```
Comment (bedeoverend) [inline src/simpla-collection-editor.html:339]: @seaneking didn't know if you knew this one, but if capture an event from inside a dom-repeat it'll have a `model` property which will have a bunch of handy properties, including the index of the item that was clicked. Also the item itself, so I could have also done: `let itemToRemove = e.model.item`

File: src/simpla-collection.html (hunk 1 above)
Comment (madeleineostoja) [reply src/simpla-collection.html:48]: Bah, keep forgetting that. Stupid Polymer, wish it was stricter one way or the other.

File: src/simpla-collection-editor.html (hunk 2 above)
Comment (madeleineostoja) [reply src/simpla-collection-editor.html:339]: Ahh nice, no didn't know that

**Comment by madeleineostoja** [comment]
> Also removes the add / remove / swap functions on the view collection. @seaneking what do you think of this?

That was the whole reason I sent events from the editor to the mediator. It should be the one to add/remove/swap, the editor should just call methods on the actual element. Otherwise it gets real messy and ugly, IMO.

**Comment by madeleineostoja** [comment]
What was wrong with sending events and letting the mediator handle it? That's the point of a mediator after all, then the editor is just a dumb UI component, I thought it was pretty clean.

**Comment by madeleineostoja** [comment]
LGTM 👍

**Pull request merged**